From 39425cc2d98801d2f0f2fed2ba01781e20d978a2 Mon Sep 17 00:00:00 2001
From: Ramprasad N <x0038811@ti.com>
Date: Thu, 18 Jan 2018 17:01:37 +0530
Subject: [PATCH 2/5] parsers: Pick bug fixes on different parsers

The following bug fixes which were previously identified are picked :
1. Increase rank for vc1parse and jpegparse
2. Modify h264 and mpeg4 parsers to interpolate pts

Signed-off-by: Pooja Prajod <a0132412@ti.com>
Signed-off-by: Ramprasad N <x0038811@ti.com>
Signed-off-by: Angela Stegmaier <angelabaker@ti.com>
---
 gst/jpegformat/gstjpegformat.c        | 2 +-
 gst/jpegformat/gstjpegparse.c         | 2 +-
 gst/videoparsers/gsth264parse.c       | 2 +-
 gst/videoparsers/gstmpeg4videoparse.c | 2 +-
 gst/videoparsers/gstmpegvideoparse.c  | 2 +-
 gst/videoparsers/plugin.c             | 2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/gst/jpegformat/gstjpegformat.c b/gst/jpegformat/gstjpegformat.c
index b410466..ecb9311 100644
--- a/gst/jpegformat/gstjpegformat.c
+++ b/gst/jpegformat/gstjpegformat.c
@@ -30,7 +30,7 @@
 static gboolean
 plugin_init (GstPlugin * plugin)
 {
-  if (!gst_element_register (plugin, "jpegparse", GST_RANK_NONE,
+  if (!gst_element_register (plugin, "jpegparse", GST_RANK_PRIMARY + 2,
           GST_TYPE_JPEG_PARSE))
     return FALSE;
   if (!gst_element_register (plugin, "jifmux", GST_RANK_SECONDARY,
diff --git a/gst/jpegformat/gstjpegparse.c b/gst/jpegformat/gstjpegparse.c
index 0bfdc80..b7d19b6 100644
--- a/gst/jpegformat/gstjpegparse.c
+++ b/gst/jpegformat/gstjpegparse.c
@@ -151,7 +151,7 @@ gst_jpeg_parse_class_init (GstJpegParseClass * klass)
 
   gst_element_class_set_static_metadata (gstelement_class,
       "JPEG stream parser",
-      "Video/Parser",
+      "Codec/Parser/Video",
       "Parse JPEG images into single-frame buffers",
       "Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>");
 
diff --git a/gst/videoparsers/gsth264parse.c b/gst/videoparsers/gsth264parse.c
index 7751ea2..5ce9030 100644
--- a/gst/videoparsers/gsth264parse.c
+++ b/gst/videoparsers/gsth264parse.c
@@ -161,7 +161,7 @@ static void
 gst_h264_parse_init (GstH264Parse * h264parse)
 {
   h264parse->frame_out = gst_adapter_new ();
-  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (h264parse), FALSE);
+  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (h264parse), TRUE);
   GST_PAD_SET_ACCEPT_INTERSECT (GST_BASE_PARSE_SINK_PAD (h264parse));
   GST_PAD_SET_ACCEPT_TEMPLATE (GST_BASE_PARSE_SINK_PAD (h264parse));
 
diff --git a/gst/videoparsers/gstmpeg4videoparse.c b/gst/videoparsers/gstmpeg4videoparse.c
index 38e1f45..49434c7 100644
--- a/gst/videoparsers/gstmpeg4videoparse.c
+++ b/gst/videoparsers/gstmpeg4videoparse.c
@@ -183,7 +183,7 @@ gst_mpeg4vparse_init (GstMpeg4VParse * parse)
   parse->interval = DEFAULT_CONFIG_INTERVAL;
   parse->last_report = GST_CLOCK_TIME_NONE;
 
-  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), FALSE);
+  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), TRUE);
   GST_PAD_SET_ACCEPT_INTERSECT (GST_BASE_PARSE_SINK_PAD (parse));
   GST_PAD_SET_ACCEPT_TEMPLATE (GST_BASE_PARSE_SINK_PAD (parse));
 }
diff --git a/gst/videoparsers/gstmpegvideoparse.c b/gst/videoparsers/gstmpegvideoparse.c
index 6896339..cd87838 100644
--- a/gst/videoparsers/gstmpegvideoparse.c
+++ b/gst/videoparsers/gstmpegvideoparse.c
@@ -173,7 +173,7 @@ gst_mpegv_parse_init (GstMpegvParse * parse)
 {
   parse->config_flags = FLAG_NONE;
 
-  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), FALSE);
+  gst_base_parse_set_pts_interpolation (GST_BASE_PARSE (parse), TRUE);
   GST_PAD_SET_ACCEPT_INTERSECT (GST_BASE_PARSE_SINK_PAD (parse));
   GST_PAD_SET_ACCEPT_TEMPLATE (GST_BASE_PARSE_SINK_PAD (parse));
 }
diff --git a/gst/videoparsers/plugin.c b/gst/videoparsers/plugin.c
index 2deddf4..c622b50 100644
--- a/gst/videoparsers/plugin.c
+++ b/gst/videoparsers/plugin.c
@@ -54,7 +54,7 @@ plugin_init (GstPlugin * plugin)
   ret |= gst_element_register (plugin, "h265parse",
       GST_RANK_SECONDARY, GST_TYPE_H265_PARSE);
   ret |= gst_element_register (plugin, "vc1parse",
-      GST_RANK_NONE, GST_TYPE_VC1_PARSE);
+      GST_RANK_PRIMARY + 2, GST_TYPE_VC1_PARSE);
 
   return ret;
 }
-- 
1.9.1

